src/DurableTask.Core/Tracing/TraceHelper.cs

```diff
@@ -228,7 +228,7 @@ internal static void EmitTraceActivityForTimer(OrchestrationInstance? instance,
             }
         }
 
-        internal static void EmitActivityforTaskFinished(OrchestrationInstance? instance, TaskScheduledEvent taskScheduledEvent, TaskFailedEvent? taskFailedEvent = null)
+        internal static void EmitActivityforTaskFinished(OrchestrationInstance? instance, TaskScheduledEvent taskScheduledEvent, TaskFailedEvent? taskFailedEvent = null, ErrorPropagationMode errorPropagationMode = ErrorPropagationMode.SerializeExceptions)
```

Comment:
I suggest we not have a default value for this and always be explicit.
```suggestion
        internal static void EmitActivityforTaskFinished(OrchestrationInstance? instance, TaskScheduledEvent taskScheduledEvent, ErrorPropagationMode errorPropagationMode, TaskFailedEvent? taskFailedEvent = null)
```

Reply:
I made it an optional parameter because it is only used if the `TaskActivity` or sub-orchestration fails. This parameter is unnecessary when they complete successfully so I refactored the code a little bit. I created separate TraceHelper methods for completed and failed `TaskActivity`s and sub-orchestrations.